File: Client/models/AttributeDTO.ts
import { DataType } from '~/models/Enums/DataType';

export interface AttributeGetDTO {
  id: number;
  name: string;
  type: string;
  typeId: number;
  dataType: DataType;
  usesDefinedValues: boolean;
  usesDefinedUnits: boolean;
}

export interface AttributeDetailsGetDTO {
  id: number;
  name: string;
  type: string;
  typeId: number;
  defaultValue: string;
  defaultUnit: string | null;
  usedCount: number;
  dataType: DataType;
}

export interface AttributePostDTO {
  name: string;
  attributeTypeId: number;
}

export interface AttributePatchDTO {
  id: number;
  name: string;
  attributeTypeId: number;
}

File: Server/AppAPI/1.0/AttributeDTO.cs
using System.ComponentModel.DataAnnotations;
using AppAPI._1._0.Enums;
using DAL.Base.Entities;

namespace AppAPI._1._0
{
    public class AttributeGetDTO : DomainEntityId
    {
        public string Name { get; set; } = default!;
        public string Type { get; set; } = default!;
        
        public long TypeId { get; set; }
        
        public AttributeDataType DataType { get; set; }
        
        public bool UsesDefinedValues { get; set; }
        public bool UsesDefinedUnits { get; set; }
    }

    public class AttributeDetailsGetDTO : DomainEntityId
    {
        public string Name { get; set; } = default!;
        public string Type { get; set; } = default!;
        
        public long TypeId { get; set; }
        
        public string? DefaultValue { get; set; }
        public string? DefaultUnit { get; set; }
        
        public long UsedCount { get; set; }
        
        public AttributeDataType DataType { get; set; }
    }

    public class AttributePostDTO
    {
        [Required] public string Name { get; set; } = default!;
        [Required] public long AttributeTypeId { get; set; } = default!;
    }

    public class AttributePatchDTO : DomainEntityId
    {
        [Required] public string Name { get; set; } = default!;
        public long? AttributeTypeId { get; set; }
    }
}